src/index.js

```diff
@@ -1 +1,34 @@
-export default () => true
+import defaultsDeep from 'lodash.defaultsdeep';
+import pickBy from 'lodash.pickby';
+import makeOkCheck from './make-ok-check';
+import buildOptions from './build-options';
+import resolveUrl from './resolve-url';
+
+class Stipulate {
+
+  constructor(options) {
+    this.baseOptions = options;
+  }
+
+  beforeRequest(url, options) {
+    return [url, options];
+  }
+
+  afterResponse(response) {
+    return response;
+  }
+
+  send(urlString, options, query) {
+    const config = buildOptions(options, this.baseOptions);
+    const url = resolveUrl(urlString, query);
+
+    const requestArguments = this.beforeRequest(url, config);
+    const checkOk = makeOkCheck(config);
+
+    return fetch(...requestArguments)
+      .then(checkOk)
+      .then(this.afterResponse);
```

Comment:
Eh.... I guess it's not needed, but the context of `afterResponse`  is the promise unless it's binded. It strikes me as wrong to be calling `this.x` when the context of the function won't be executed in the context of `this`.

Reason: These functions are designed to be extended, and since it's a class, it's reasonable to assume that the extended functions are executed in the context of the object and that `this.x` is available from within the function.

Say I want to attach a new property to the class, `ExtStip.foo` and I want to do something with it in the `afterResponse` hook. I can't, unless I also extend the `send` method or do some unintuitive bindings in the `afterResponse`. 


By design, `send` should never be extended. You're giving functions that are executed before/after send does it's thing, and if someone does need their own version of `send`, they probably don't want Stipulate to begin with.

Maybe expecting the class to be extended is also the wrong approach here. Maybe we should be providing hooks instead.

```js
...
  _afterResponse(response) {
     return (response) => {
        if (this.afterResponse && typeof this.afterResponse === 'Function') {
          return this.afterResponse(response);
        }
      
        return response;
     }();
  }
...
```

Actually, I think we should do this approach either way. `beforeRequest` and `afterResponse` can be set during an extension or on the fly (I might not want to create a subclass just to access hooks), and the decision is up to the developer.